### ISSUE TITLE
Non-owners should be able to moderate channels

### ISSUE THREAD

**Issue by holmesworcester** [TryQuiet/quiet#1536]
Right now there are just two types of user: the community creator (owner) and everyone else. This will be impractical for moderation. 

The community creator should be able to appoint others who can do things like remove users and hide messages. 

We have some designs for this but they probably need updating for mobile.

**Comment by MrLuxuri** [comment]
Right, and for that we need to bring "role creation" like Discord/Guilded, so I can make a moderator role, choose a set of permissions from a list and then give that role to those who I wish to help me Moderate, I'd still own my server as admin, but these guys will help me as mods.

**Comment by holmesworcester** [comment]
Yeah! Our current plan is to do roles similar to how Discord does them.

One question: do you have any complaints about settings roles or adding/promoting mods on mobile? We'd like to use just one set of designs so we don't have big differences between desktop and mobile.

**Comment by holmesworcester** [comment]
@MrLuxuri what do you think about these designs for roles?

https://www.figma.com/file/xCgVHg3qZPCQuLZudBAnn7/Roles?type=design&node-id=563-10326&mode=design&t=KnB10fFchg6Lzdqa-0

Clickable prototypes are here:

https://www.figma.com/proto/xCgVHg3qZPCQuLZudBAnn7/Roles?type=design&node-id=537-5581&t=KnB10fFchg6Lzdqa-0&scaling=min-zoom&page-id=7%3A2510&starting-point-node-id=537%3A5581&show-proto-sidebar=1 (There are four different flows you can go through in the left sidebar.)

**Comment by holmesworcester** [comment]
@jgaylor do we have designs for removing users or silencing them?

**Comment by jgaylor** [comment]
@holmesworcester 

> do we have designs for removing users or silencing them?

Yes, 
Users with permission to remove can do this 2 ways:

1. Remove members from a role by going to Role > Members. [Figma](https://www.figma.com/file/xCgVHg3qZPCQuLZudBAnn7/Roles?type=design&node-id=537%3A8483&mode=design&t=LKipFC0KJ6vgR5Rn-1).
2. Remove members from a role by going to Profile > Edit roles. [Figma](https://figma.com/file/xCgVHg3qZPCQuLZudBAnn7/Roles?type=design&node-id=534%3A9381&mode=design&t=LKipFC0KJ6vgR5Rn-1)
